sympy/polys/tests/test_hypothesis.py

```diff
@@ -1,17 +1,80 @@
 from hypothesis import given
-from hypothesis import strategies as st
-from sympy.abc import x
-from sympy.polys.polytools import Poly
+from sympy.testing.hypothesis import polys, lattice_axioms_singular, lattice_axioms_dual
+from sympy.polys.polytools import lcm, gcd
+import pytest
 
 
-def polys(*, nonzero=False, domain="ZZ"):
-    # This is a simple strategy, but sufficient the tests below
-    elems = {"ZZ": st.integers(), "QQ": st.fractions()}
-    coeff_st = st.lists(elems[domain])
-    if nonzero:
-        coeff_st = coeff_st.filter(any)
-    return st.builds(Poly, coeff_st, st.just(x), domain=st.just(domain))
 
+@given(
+    f=polys(),
+    g=polys(),
+    h=polys(),
+)
+@pytest.mark.xfail
+def test_gcd(f, g, h):
+    assert lattice_axioms_singular([f, g, h], gcd)
+
+    # multiply by h
+    gcd_h = g.gcd(f + h * g)
+    assert f.gcd(g) == gcd_h
+
+
+@given(f=polys(), g=polys(), h=polys())
+@pytest.mark.xfail
+def test_lcm(f, g, h):
+    assert lattice_axioms_singular([f, g, h], lcm)
+    assert f * g == f.lcm(g) * f.gcd(g)
+
+
+@given(f=polys(), g=polys(), h=polys())
+@pytest.mark.xfail
+def test_lcm_gcd(f, g, h):
+    assert lattice_axioms_dual([f, g, h], [lcm, gcd])
+
+
+@given(
+    f=polys(),
+    g=polys(empty=False),
+    h=polys(domain="QQ"),
+    l=polys(empty=False, domain="QQ"),
+)
+def test_division(f, g, h, l):
+    # Integer case
+    z = f.rem(g)
+    assert g.degree() >= z.degree() or z.degree() == 0
+
+    # Rational case
+    q = h.rem(l)
+    assert l.degree() >= q.degree() or q.degree() == 0
```

Comment:
For the rational case this should be a strict inequality:
```python
assert l.degree() > q.degree()
```
The integer case apparently converts tor rational depending on whether `auto=True` so
```python
z1 = f.rem(g, auto=True)
z2 = f.rem(g, auto=False)
assert z1.degree() < g.degree()
assert z2 == f or z2.degree() < f.degree()
```